### DISCUSSION TITLE
Support TIFFs utilizing ExtraSamples

### DISCUSSION THREAD

**Discussion by dvdgutzmann** [SixLabors/ImageSharp#2026] [Ideas] [answered]
First of all, I want to tell you how much we appreciate your work. ImageSharp is a great library and we have never had a problem with it. We recently upgraded to v2 to take advantage of the TIFF support. 

When loading some TIFFs via `Image.Load(stream, out var imageFormat)`, an exception is thrown stating _ExtraSamples is not supported_ . We understand that ExtraSamples are a way to achieve transparency in TIFFs.

**Comment by brianpopow** [comment]
As the exception states, we do not support ExtraSamples yet. You can find an overview of what is supported and what not in the ReadMe in the tiff format folder: [TiffReadMe](https://github.com/SixLabors/ImageSharp/blob/main/src/ImageSharp/Formats/Tiff/README.md)

Any help with implementing missing features is very welcome.

**Reply by dvdgutzmann** [reply]
Thank you for pointing out the ReadMe. Unfortunately, image processing is not exactly our field of expertise, we probably wouldn't be much help.

**Reply by JimBobSquarePants** [reply]
It's basically a case of read the spec, implement the spec when it comes to implementing image formats.

**Comment by brianpopow** [comment, marked as answer]
With PR #2062, #2047 and #2037 merged to main, we support now decoding Tiff images with alpha data. It can be tried out via our [nightly builds](https://www.myget.org/feed/sixlabors/package/nuget/SixLabors.ImageSharp).

edit: with 2.1.0 released, its now available via the nuget feed
